Code-Sleep-Python/Prime/sieve_of_eratosthenes.py

```diff
@@ -0,0 +1,157 @@
+    pygame.display.update()
+
+
+celldict = {}  # dictionary to map numbers to their respective cells
+
+
+class Cell:  # creating a class for Cell objects on the grid
+    def __init__(self, x, y, n):
+        self.pos = (x, y)
+        self.value = n
+        self.highlighted = False
+        # adding the number and cell pair to the celldict dictionary
+        celldict[self.value] = self
+        # display the value of the cell on the grid at the center of the cell
+        message_display(f"{self.value}", x + 16, y + 16, 17)
+
+    def highlight(self):
+        if not self.highlighted:
+            x, y = self.pos
+            # create a MAUVE rectangle on top of the cell to highlight
+            # composite numbers
+            pygame.draw.rect(display, MAUVE, (x, y, 32, 32))
+            # rewrite the value of the cell on top of the MAUVE highlight
+            message_display(f"{self.value}", x + 16, y + 16, 17)
+            # to avoid time to re-highlight cells that have already been
+            # highlighted
+            self.highlighted = True
+            # for the effect of moving highlight or "scanning" the grid
+            pygame.time.wait(50)
+
+
+count = 1
+for i in range(0, width, 32):
+    # to deal with internal PyGame events in the event queue which cause the
+    # system to freeze
+    pygame.event.pump()
+    for j in range(0, width, 32):
+        if (count <= num):
+            # create a Cell object at every grid intersection
+            cell = Cell(j, i, count)
+            clock.tick()
+            count += 1
+        else:
+            break
+pygame.display.update()
+# Sieving algorithm
+# all the numbers except 1 since 1 is neither prime nor composite
+nums = set(range(2, num + 1))
+composites = {1}  # set of composites detected so far
+celldict[1].highlight()  # highlight the Cell 1
+
+for i in range(
+        2,
+        round(
+            math.sqrt(num)) +
+        1):  # check all numbers till sqrt(num)
+    if i not in composites:
+        # cross out all the multiples of prime numbers
+        comp = [i * k for k in range(2, num // i + 1)]
+        composites.update(comp)
+        for val in comp:
+            pygame.event.pump()
+            celldict[val].highlight()  # highlight the composites
+
+primetxt = ''  # string of primes to be displayed
+for val, cell in celldict.items():
+    pygame.event.pump()
+    if not cell.highlighted:  # if the cell has not been highlighted
+                              # it is prime
+        x, y = cell.pos
+        # highlight the primes with GREEN
+        pygame.draw.rect(display, GREEN, (x, y, 32, 32))
+        message_display(f"{val}", x + 16, y + 16, 17)
+        pygame.time.wait(100)  # to create a moving highlight
+        primetxt += str(val) + ' ,'  # concatenate primes to primetxt string
+
+# Displaying results using GUI
+m = tkinter.Tk()  # root window
+l1 = tkinter.Label(m, text='Primes that were sieved: ')
+# Message widget to display the sieved primes

```

Comment:
Rearrange code in a more well-fashioned way.
First have all class definitions and methods and normal functions.

Then have the main code [those without any indent]